### ISSUE TITLE
bug when using overload decorator and numpy arrays

### ISSUE THREAD

**Issue by martinResearch** [astral-sh/ty#421]
### Summary

I am getting some error when using a combination of overload and numpy arrays while the code runs and mypy does not complain:

```py
import numpy as np
from typing import overload
 
@overload
def f( a:  np.ndarray, index: int) -> int: ...
@overload
def f( a:  np.ndarray, index: np.ndarray ) -> np.ndarray:  ...
def f(a:  np.ndarray, index: int | np.ndarray) -> np.ndarray|int:
    if index is int:
        return a[index]
    elif isinstance(index,  np.ndarray):
        return a[index]
    else:
        raise TypeError("index must be int or list[int]")


def g(a:  np.ndarray, index:  np.ndarray) -> np.ndarray:
    return a[index]

@overload
def h( a:   list[int], index: int) -> int: ...
@overload
def h( a:   list[int], index: list[int]) -> list[int]:
   ...
def h(a:   list[int], index: int | list[int]) -> list[int]|int:
    if index is int:
        return a[index]
    elif isinstance(index, list):
        return [a[i] for i in index]
    else:
        raise TypeError("index must be int or list[int]")

def main()->None:
    a=np.zeros((10))
    i=np.array([4,5])
    g_a = g(a, i) 
    f_a=f(a,i)
    h_a=h(a,[4,5])
    print(len(h_a)) # works fine
    print(len(g_a)) # works fine
    print(len(f_a)) # Fails: Expected `Sized`, found `int`
    
if __name__ == "__main__":
    main()
```
is that expected? Maybe the problem lies on numpy's side?

### Version

ty==0.0.1a3

**Comment by sharkdp** [comment]
> is that expected?

No, thank you for reporting this!

The problem is that we currently do not understand the return type of `np.array([4, 5])`, so `i` ends up being some `@Todo` type, which signals a missing feature (in this case: support for `typing.TypeAlias`). Those `@Todo` types are assignable to everything, which is why we pick the first `f` overload here, returning `int` instead of `np.ndarray`.
